app/src/main/java/com/pmatuki/wowpapers/view/detail/DetailActivity.kt

```diff
@@ -37,9 +35,7 @@ class DetailActivity : AppCompatActivity() {
 
     private fun bindViewModel() {
         viewModel = ViewModelProvider(
-            this, DetailViewModelFactory(
-                ImageDownloadService(applicationContext), WallpaperService(applicationContext)
-            )
+            this, DetailViewModelFactory()
```

Comment:
Although it is correct, I prefer the approach to create an extension to encapsulate this logic, similar to the one I shared on the [sample code](https://github.com/villela/toothpickviewmodelexample/blob/master/app/src/main/java/com/matheusvillela/toothpickviewmodelexample/AppExtensions.kt).

IMHO, the View should know almost nothing about ViewModelProviders and which Factory needs to be called. Dor the View, it only matters that it should receive the ViewModel.

Reply:
Will do!